common/crypto.hpp

```diff
@@ -39,34 +41,67 @@ enum CryptoErrorCode {
 	VerificationError,
 };
 
+struct Args {
+	const string private_key_path;
+	const string private_key_passphrase;
+	const string ssl_engine;
+};
```

Comment:
Good thinking!

Reply:
I am also pondering typing some of these, so that it will be harder to mess up the argument order, but might be I'm going overboard here...